feat: Support dynamic container registry mirror

yxf0314 opened the pull request:
https://github.com/gpustack/gpustack/issues/3159

The functional logic is as described in the comment:
```
1) If the image has an explicit registry, return it as is.
2) If the image does not have an explicit registry and a system default registry is configured,
   prefix the image with the system default registry in config.
3) If the image does not have an explicit registry and no system default registry is configured,
   using docker.io as default if image without "gpustack" prefix.
4) If the image does not have an explicit registry and no system default registry is configured,
   and with "gpustack" prefix, using docker.io as default if docker.io is reachable. Otherwise, using quay.io.
```

gitlawr commented:
1. Can we do the check once on bootstrap?
2. How can users update the registry without restarting gpustack?

gitlawr commented on gpustack/worker/inference_backend_manager.py:
```diff
@@ -119,3 +122,20 @@ def _handle_event(self, event: Event):
+            logger.debug(f"Docker Hub reachable: {_is_docker_hub_reachable}")
+        except Exception as e:
+            _is_docker_hub_reachable = False
+            logger.warning(f"Docker Hub connectivity check failed: {e}")
```
log the error in debug. Use info to tell users Docker Hub connectivity check failed and we are gonna switch to use quay.io

yxf0314 commented:
> 1. Can we do the check once on bootstrap?
> 2. How can users update the registry without restarting gpustack?

1. Moved the reachability check to the startup phase. However, since the inference backend starts in a separate process, global variables need to be reassigned during process creation.
2. Added a reload-config command, user can run like `gpustack reload-config --config-file /path/file.yaml --system-default-container-registry test-registry.io` to reload config from file or args

gitlawr commented on gpustack/main.py:
```diff
@@ -19,10 +20,12 @@ def main():
     )
 
     setup_start_cmd(subparsers)
+    setup_reload_config_cmd(subparsers)
```
Add a cli reference doc for it
